include/cassandra.h

```diff
@@ -1515,203 +1509,428 @@ cass_session_get_schema(CassSession* session);
- * @see cass_schema_meta_type()
- * @see cass_iterator_from_schema_meta()
- * @see cass_iterator_fields_from_schema_meta()
+ * @param[in] keyspace_meta
+ * @param[in] name
+ * @return A schema metadata field. NULL if the field does not exist.
  */
-CASS_EXPORT const CassSchemaMeta*
-cass_schema_meta_get_entry(const CassSchemaMeta* meta,
-                           const char* name);
+CASS_EXPORT const CassMetaField*
+cass_keyspace_meta_field_by_name(const CassKeyspaceMeta* keyspace_meta,
+                                 const char* name);
 
 /**
- * Same as cass_schema_meta_get_entry(), but with lengths for string
+ * Same as cass_keyspace_meta_field_by_name(), but with lengths for string
  * parameters.
  *
- * @public @memberof CassSchema
+ * @public @memberof CassKeyspaceMeta
  *
- * @param[in] meta
+ * @param[in] keyspace_meta
  * @param[in] name
  * @param[in] name_length
- * @return same as cass_schema_meta_get_entry()
+ * @return same as cass_keyspace_meta_field_by_name()
  *
- * @see cass_schema_meta_get_entry()
+ * @see cass_keyspace_meta_field_by_name()
  */
-CASS_EXPORT const CassSchemaMeta*
-cass_schema_meta_get_entry_n(const CassSchemaMeta* meta,
-                             const char* name,
-                             size_t name_length);
+CASS_EXPORT const CassMetaField*
+cass_keyspace_meta_field_by_name_n(const CassKeyspaceMeta* keyspace_meta,
+                                   const char* name,
+                                   size_t name_length);
+
+/**
+ * Gets the column metadata for the provided column name.
+ *
+ * @public @memberof CassTableMeta
+ *
+ * @param[in] table_meta
+ * @param[in] column
+ *
+ * @return The metadata for a column. NULL if column does not exist.
+ */
+CASS_EXPORT const CassColumnMeta*
+cass_table_meta_column_by_name(const CassTableMeta* table_meta,
+                               const char* column);
+
+/**
+ * Same as cass_table_meta_column_by_name(), but with lengths for string
+ * parameters.
+ *
+ * @public @memberof CassTableMeta
+ *
+ * @param[in] table_meta
+ * @param[in] column
+ * @param[in] column_length
+ * @return same as cass_table_meta_column_by_name()
+ *
+ * @see cass_table_meta_column_by_name()
+ */
+CASS_EXPORT const CassColumnMeta*
+cass_table_meta_column_by_name_n(const CassTableMeta* table_meta,
+                                 const char* column,
+                                 size_t column_length);
+
+/**
+ * Gets the name of the table.
+ *
+ * @public @memberof CassTableMeta
+ *
+ * @param[in] table_meta
+ * @param[out] name
+ * @param[out] name_length
+ */
+CASS_EXPORT void
+cass_table_meta_name(const CassTableMeta* table_meta,
+                     const char** name,
+                     size_t* name_length);
+
+/**
+ * Gets the id of the table.
+ *
+ * @public @memberof CassTableMeta
+ *
+ * @param[in] table_meta
+ * @return The UUID id of the table.
+ */
+CASS_EXPORT CassUuid
+cass_table_meta_id(const CassTableMeta* table_meta);
```

Comment:
For C* v1.2.x an 2.0.x `cf_id` does not exist resulting in the return of a uninitialized id_ (https://github.com/datastax/cpp-driver/pull/233/files#diff-f30001b0d6d7abd11f9acc768a8b0737R240); the CassUuid struct does not define a constructor to to initialize either. Should we define a CASS_UUID_INVALID in the API header and initialize CassUuid as the invalid define?

Or we could document that the meta id is only available for C* 2.1+.